cmd/lncli/commands.go

```diff
@@ -2259,40 +2279,73 @@ func sendPaymentRequest(ctx *cli.Context, req *lnrpc.SendRequest) error {
 		}
 	}
 
-	paymentStream, err := client.SendPayment(context.Background())
+	feeLimit, err := retrieveFeeLimit(ctx, amt)
 	if err != nil {
 		return err
 	}
+	req.FeeLimit = feeLimit
 
-	if err := paymentStream.Send(req); err != nil {
+	_, err = routerClient.SendPayment(context.Background(), req)
+	if err != nil {
 		return err
 	}
 
-	resp, err := paymentStream.Recv()
+	fmt.Printf("Payment in flight, track status by running:\n")
+	fmt.Printf("lncli lookuppayment %v\n", hash)
+
+	return nil
+}
+
+var lookupPaymentCommand = cli.Command{
+	Name:      "lookuppayment",
+	Category:  "Payments",
+	ArgsUsage: "hash",
+	Action:    actionDecorator(lookupPayment),
+}
+
+func lookupPayment(ctx *cli.Context) error {
+	args := ctx.Args()
+
+	conn := getClientConn(ctx, false)
+	defer conn.Close()
+
+	client := routerrpc.NewRouterClient(conn)
+
+	if !args.Present() {
+		return fmt.Errorf("hash argument missing")
+	}
+
+	hash, err := hex.DecodeString(args.First())
 	if err != nil {
 		return err
 	}
 
-	paymentStream.CloseSend()
-
-	printJSON(struct {
-		E string       `json:"payment_error"`
-		P string       `json:"payment_preimage"`
-		R *lnrpc.Route `json:"payment_route"`
-	}{
-		E: resp.PaymentError,
-		P: hex.EncodeToString(resp.PaymentPreimage),
-		R: resp.PaymentRoute,
-	})
+	req := &lnrpc.PaymentHash{
+		RHash: hash,
+	}
 
-	// If we get a payment error back, we pass an error
-	// up to main which eventually calls fatal() and returns
-	// with a non-zero exit code.
-	if resp.PaymentError != "" {
-		return errors.New(resp.PaymentError)
+	stream, err := client.LookupPayment(context.Background(), req)
+	if err != nil {
+		return err
 	}
 
-	return nil
+	for {
```

Comment:
The help for the command should detail that this command may not return immediately. Also we may want to consider a mode where simply prints the latest state and exists early. In this case, something along the lines of a `--watch` flag would be added to allow the user to request on going notifications.

Reply:
Not applicable if we leave `lncli unchanged`.